### PR TITLE
Add higher impact rules from okta-ui-react

### PR DESCRIPTION
**Pull request by michaeltamaki-okta**
**Rebased on top of #1, so some commits are from that PR**

Add higher impact rules from okta-ui-react that may require more discussion.

See wiki for individual rule reasonings: https://oktawiki.atlassian.net/l/c/xcb8tDzu

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -91,15 +95,15 @@ module.exports = {
...
     "prefer-template": "error",
-    "quotes": "off",
+    "quotes": ["error", "single"],
     "react-hooks/rules-of-hooks": "error",
     "react/no-danger": "error",
     "react/no-deprecated": "error",
     "react/no-multi-comp": "off",
     "react/no-string-refs": "warn",
     "react/no-unsafe": ["error", { "checkAliases": true }],
     "react/prop-types": "off",
```
Comment (ghost) [inline index.js]: unrelated to your changes, but it is a higher impact rule change suggestion!

@mattkaniaris-okta is there a particular reason for keeping this `off`?

I can't think of a good reason to skip prop types for react components.

**Comment by ggreer** [comment]
This eslint config is used in ASA, Workflows, and in both frontend and node.js code. All of those codebases are *very* different, and changing them all to satisfy stricter eslint configs would be a giant pain in some cases, so we picked our battles.

If these changes are merged, ASA & Workflows will never be able to use newer versions of this eslint config.

I think the solution here is to create a strict.js and put the stricter rules in it. It can extend the original eslint config to reduce repetition. Then in okta-ui-react, you'd have:

```
"extends": [
     "eslint-config-okta/strict",
  ],
```

See https://eslint.org/docs/developer-guide/shareable-configs#sharing-multiple-configs for more info on how this works.

Edit: To be clear, if I could snap my fingers and have all the codebases obey these rules, I totally would. It's just that not all of these rules can be fixed with `eslint --fix` and there is a ton of code in Workflows that would need to be changed. That's why I think a strict set of rules would be better. All new repos and okta-ui-react could use the strict rules right off the bat, and we could migrate workflows repos over one at a time.

File: index.js (hunk 1 above)
Comment (ggreer) [reply index.js]: Workflows has hundreds of react components and many of them lack propTypes. :(

IIRC in ASA, it's mostly TypeScript so there's little point in having proptypes.

**Review by ggreer — APPROVED** [review]
LGTM. Nice work.

**Review by jmelberg-okta — APPROVED** [review]
Looks good! Before merging, can you add the `strict` example to the [README](https://github.com/oktadeveloper/eslint-config#usage)?

As a follow-up, let's create another PR with a version bump so we can start using it! 😄

**Pull request merged**